src/main/controllers/claimTaskController.ts

```diff
@@ -1,26 +1,37 @@
 import { Request, Response } from 'express';
 import Debug from 'debug';
 import { Task } from '../models/task';
+import { MyCasesPage } from '../models/myCasesPage';
 
 const claimTaskDebug = Debug('app:controller:claimTask');
+const stages = require('../data/stages');
 
 export function claimTask(req: Request, res: Response): void {
   claimTaskDebug(`myCaseWork.claimTask controller with caseRef=${req.query.caseRef}...`);
   const filteredAvailableTasks: Array<Task> = req.session.myFilteredAvailableTasks;
   const myAvailableTasks: Array<Task> = req.session.myAvailableTasks;
+  const complete = req.query.complete;
+  if (complete === 'true') {
+    req.session.myTasks = req.session.myTasks.filter((task: Task) => task.caseRef !== req.query.caseRef);
+    req.session.myFilteredAvailableTasks = filteredAvailableTasks.filter(x => x.caseRef !== req.query.caseRef);
```

Comment:
you also need to update the available object: 
req.session.myAvailableTasks = myAvailableTasks.filter(x => x.caseRef !== req.query.caseRef);

Reply:
otherwise, the filter will stop working properly